pyro/infer/tracegraph_elbo.py

```diff
@@ -92,20 +137,30 @@ def _compute_downstream_costs(model_trace, guide_trace,  #
 
 
 def _compute_elbo_reparam(model_trace, guide_trace):
+
+    # Compute a surrogate ELBO. In [1], this is simply the ELBO because
```

Comment:
i don't really understand this comment. in ref [1] they also make use of the language of a surrogate loss function (section 3.2)

Reply:
A more precise start to my comment might be: 

In ref [1], the portion of the surrogate loss handled by this function is `\sum{c}`, which in this context would simply be the ELBO...

Reply:
Thanks for engaging with this! Realized that if my image were right, the implementation would be wrong, and that has been a very useful signal to revisit my understanding. Would this be closer to the truth?

    # In ref [1], the part of the surrogate loss computed here is `\sum{c}`, 
    # which in this case is simply the ELBO. Depending on the parameterization, 
    # some of the entropy terms may not contribute (in expectation) to the 
    # gradient estimate. To reduce the variance of the gradient estimates under some
    # conditions, the log q(z|...) terms are instead taken from `site['score_parts']`

Reply:
@martinjankowiak marking these as (hopefully) resolved to simplify reviewing the new commit